.eslintrc.js

```diff
@@ -71,6 +71,16 @@ module.exports = {
       'error',
       'consistent'
     ],
+    'object-shorthand': 0,
+    'n/no-callback-literal': 0,
+    '@typescript-eslint/await-thenable': 0,
+    '@typescript-eslint/consistent-generic-constructors': 0,
+    '@typescript-eslint/ban-ts-comment': 0,
+    '@typescript-eslint/consistent-indexed-object-style': 0,
+    '@typescript-eslint/no-confusing-void-expression': 0,
+    '@typescript-eslint/ban-types': 0,
+    '@typescript-eslint/consistent-type-imports': 0,
+    '@typescript-eslint/consistent-type-exports': 0,
```

Comment:
Are none of these automatically fixable, or was some of this disabling a style preference?

Reply:
e.g. `object-shorthand` seems to default to `always` and says "some" things are fixable with `--fix`. Just checking if they are disabled because they couldn't be fixed with `--fix`?

Reply:
Some of them aren't automatically fixable but to be honest, most of them are just style preferences.

----------------------------------

[`object-shorthand`](https://eslint.org/docs/latest/rules/object-shorthand) wants 
```ts
const f = {
    foo,
}

// instead of
const f = {
    foo: foo
}
```
syntax for functions - it could probably be enabled

------------------------------------

[`n/no-callback-literal`](https://github.com/bizob2828/eslint-plugin-n/blob/master/docs/rules/no-callback-literal.md) disallows callbacks called `cb` or `callback` where the first argument is not `null` `undefined` or an error. We definitely don't want this.

------------------------------------

[`@typescript-eslint/await-thenable`](https://typescript-eslint.io/rules/await-thenable) disallows awaiting non-thenable things. I actually started trying to fix this rule (it's not autofixable) but we have a bunch of `await dispatch` calls which are sometimes promises sometimes not, and I feel like it's safer if we always await them.

------------------------------------

[`@typescript-eslint/consistent-generic-constructors`](https://typescript-eslint.io/rules/consistent-generic-constructors) forces generic args to be specified on the right:
```
const foo: Bar<number> = new Bar()
// has to become
const foo: Bar<number> = new Bar<number>()
```
we could probably turn this on, but it doesn't seem that important to me as
```
const foo = new Bar()
```
will be an error anyway.

-----------------------------------------

[`@typescript-eslint/ban-ts-comment`](https://typescript-eslint.io/rules/ban-ts-comment) bans `// disable-ts` comments unless you have at least 3 characters explaining why you did it. I'd like to enable it but there are a lot of `// disable-ts` comments and I don't have context on why (for the most part).

---------------------------------------

[`@typescript-eslint/consistent-indexed-object-style`](https://typescript-eslint.io/rules/consistent-indexed-object-style) bans `{ [key: string]: WhatEver }` in favor of `Record<string, WhatEver>`. We could enable it, but tbh I don't really care either way. You can also force it to use the `{ [key: string]: WhatEver }` syntax and ban the record but both seem fine to me.

--------------------------------------

[`@typescript-eslint/no-confusing-void-expression`](https://typescript-eslint.io/rules/no-confusing-void-expression) bans braceless arrow functions that return void (I think mostly in promises). It would make me very sad if we enabled this.

```ts
promise.then(value => window.postMessage(value));
// becomes
promise.then(value => {
    window.postMessage(value)
})
```

------------------------------------

[`@typescript-eslint/ban-types`](https://typescript-eslint.io/rules/ban-types) bans a bunch of types like `{}` `Number` `Function` `String` and the like. We could probably enable it but it seems pretty unimportant

-----------------------------------

[`@typescript-eslint/consistent-type-imports`](https://typescript-eslint.io/rules/consistent-type-imports) forces imports which are only used as types to be imported via `import type`. We could probably enable this, but personally, I find it a bit annoying, especially when I need to change back and forth as I use/don't use a type as a value.

---------------------------------

[`@typescript-eslint/consistent-type-exports`](https://typescript-eslint.io/rules/consistent-type-exports) is the same but for exports. Basically it means you need to export types with `export type MyInterface` instead of `export MyInterface`. Apparently it can improve bundling speed? Honestly though, not a fan :D

Reply:
Ok I agree with pretty much all your points!